lib/handler/file.c

```diff
@@ -282,6 +381,23 @@ static int send_dir_listing(h2o_req_t *req, const char *path, size_t path_len, i
     return 0;
 }
 
+size_t *process_range(h2o_mem_pool_t *pool, h2o_iovec_t *range_value, size_t file_size, size_t* ret);
```

Comment:
Please change the meaning of the last variable to _the number of ranges found, if the function succeeds in parsing the input_, from _the number of ranges found, or error code if it fails_.

I believe that the design of the function has been borrowed from the functions of picohttpparser.  But in the case of this function we do not need to distinguish parse errors and empty input (or incomplete input).

The caller of the function can (and does) simply ignore the result if the function returns `NULL`, so changing the meaning of the last argument as suggested will not have any side effects.  OTOH it would simplify the design of the function.